src/rpc/mining.cpp

```diff
@@ -525,6 +525,7 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
         // Need to update only after we know CreateNewBlock succeeded
         pindexPrev = pindexPrevNew;
     }
+    assert(pindexPrev);
```

Comment:
I believe the reason this is valid is that we've checked `IsInitialBlockDownload()` is false, and that ensures `chainActive->Tip()` is not nullptr, and we've just ensured `pindexPrev` is `chainActive->Tip()`. We did potentially release `cs_main` in the meantime, though. I wonder if asserting `chainActive->Tip() != nullptr` or `!IsInitialBlockDownload()` would be enough to satisfy to the static analysis; either seems like it would be easier to understand, to me?

Reply:
@ajtowns On what line do you suggest placing that assertion?

Reply:
If it satisfied the analysis tool, I'd probably put `assert(chainActive->Tip() != nullptr);` somewhere around `static CBlockIndex* pindexPrev;`, I guess.